csharp/extractor/Semmle.Extraction.CIL/Entities/GenericsHelper.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+
+namespace Semmle.Extraction.CIL.Entities
+{
+    internal static class GenericsHelper
+    {
+        public static TypeTypeParameter[] MakeTypeParameters(Type container, int count)
+        {
+            var newTypeParams = new TypeTypeParameter[count];
+            for (var i = 0; i < newTypeParams.Length; ++i)
+            {
+                newTypeParams[i] = new TypeTypeParameter(container, container, i);
```

Comment:
This looks weird; A type parameter that is a type parameter of itself?

Reply:
This is coming from `TypeReferenceType` without any significant change. https://github.com/github/codeql/pull/4775/files#diff-f63420edae00ed792249eaad1286b6c3ac98109803fa6842ab7d903dfafdbc9eL42

Reply:
I removed the unneeded duplicate parameter. I think it was only there because `MethodTypeParameter` has both too.